src/darwin/agent/reportcrash.js

```diff
@@ -51,35 +51,36 @@ var AppleErrorReport = ObjC.classes.AppleErrorReport;
 var CrashReport = ObjC.classes.CrashReport;
 var NSMutableDictionary = ObjC.classes.NSMutableDictionary;
 
-var crashedPid;
-var is64Bit;
-var forcedByUs;
-var logPath;
-var logFd;
-var logChunks;
-var mappedAgents;
-
-function reset() {
-  crashedPid = -1;
-  is64Bit = null;
-  forcedByUs = false;
-  logPath = null;
-  logFd = null;
-  logChunks = [];
-  mappedAgents = [];
-}
+const state = {};
```

Comment:
- Even though newer Duktape (kinda) supports `const`, we should probably stick to `var` for consistency with our existing internal agents written in ES5 syntax.
- How about naming it `threadStates`, and use `state` for the local cases? (So the variable name length relationships are proportional to their visibility.)